pkg/ebpf/compiler/compiler.go

```diff
@@ -147,7 +51,52 @@ func NewEBPFCompiler(headerDirs []string, verbose bool) (*EBPFCompiler, error) {
 			fmt.Sprintf("-isystem%s/include/generated/uapi", d),
 		)
 	}
-	ebpfCompiler.kernelCflags = cflags
+	cflags = append(cflags, fmt.Sprintf("-isystem%s", embeddedIncludePath))
+	cflags = append(cflags, "-c", "-x", "c", "-o", "-", "-")
+
+	var clangOut, clangErr, llcErr bytes.Buffer
+
+	clangCtx, clangCancel := context.WithTimeout(context.Background(), compilationStepTimeout)
+	defer clangCancel()
 
-	return ebpfCompiler, nil
+	compileToBC := exec.CommandContext(clangCtx, clangBinPath, cflags...)
+	compileToBC.Stdin = in
+	compileToBC.Stdout = &clangOut
+	compileToBC.Stderr = &clangErr
```

Comment:
Could we log the `stderr` from `clang` and `llc` as a debug log message? I think that could be helpful debugging issues.

Reply:
If `stderr` is not nil, we return it from this function, so I don't think logging it would add anything here

Reply:
We only do that if `err != nil`. I'm looking for a debug log when compilation completes successfully.

Reply:
Ah, I see. We can definitely do that.